trustmanager/pkcs11/common/universal.go

```diff
@@ -0,0 +1,184 @@
+// +build pkcs11
+
+package common
+
+import (
+	"fmt"
+
+	"github.com/miekg/pkcs11"
+	"github.com/sirupsen/logrus"
+	"github.com/theupdateframework/notary"
+	"github.com/theupdateframework/notary/trustmanager"
+	"github.com/theupdateframework/notary/tuf/data"
+)
+
+// an interface around the pkcs11 library, so that things can be mocked out
+// for testing
+const ecdsaPrivateKeySize = 32
+
+var (
+	hardwareName     string
+	hardwareKeyStore HardwareSpecificStore
+)
+
+// SetKeyStore sets up the to be used keystore
+func SetKeyStore(ks HardwareSpecificStore) {
```

Comment:
This appears to be restricting us to only being able to access a single hardware device at a time. Is that correct? If yes, how can we not incur that limitation.

Reply:
yes this is right, depending on which libs are found (opencryptoki  and yubikey right now) a hardware device is chosen. Yet it would choose one of the libs and performing every crypto operation with it. What would be your suggestion for breaking down this limitation? And what do you mean with "at a time"? You would like to have a selection, which hardware should bes used? Didn't think about it (and it's design implications) too much yet, but with further development there could be a mechanism that chooses one if just one is present and asks which to use if more than one is found.

Reply:
Ideally we would instantiate a key store for every recognized hardware device attached to the system. You might run into cases where say, 2 yubikeys were attached, one having the root or targets key, one having your delegation key.